code/modules/supply/supply_console.dm

```diff
@@ -0,0 +1,499 @@
+	var/list/static_data = list()
+	var/list/packs_list = list()
+
+	for(var/set_name in SSeconomy.supply_packs)
+		var/datum/supply_packs/pack = SSeconomy.supply_packs[set_name]
+		if((pack.hidden && hacked) || (pack.contraband && can_order_contraband) || (pack.special && pack.special_enabled) || (!pack.contraband && !pack.hidden && !pack.special))
+			packs_list.Add(list(list(
+				"name" = pack.name,
+				"cost" = pack.cost,
+				"ref" = "[pack.UID()]",
+				"contents" = pack.ui_manifest,
+				"cat" = pack.group)))
+
+	static_data["supply_packs"] = packs_list
+
+	var/list/categories = list()
+	for(var/category in SSeconomy.all_supply_groups)
+		categories.Add(list(list("name" = get_supply_group_name(category), "category" = category)))
+
+	static_data["categories"] = categories
+
+	return static_data
+
+
+
+/obj/machinery/computer/supplycomp/ui_act(action, list/params, datum/tgui/ui)
+	if(..())
+		return
+
+	var/mob/user = ui.user
+
+	// If its not a public console, and they aint authed, dont let them use this
+	if(!is_public && !is_authorized(user))
+		return
+
+	if(!SSshuttle)
+		stack_trace("The SSshuttle controller datum is missing somehow.")
+		return
+
+	. = TRUE
+	add_fingerprint(user)
+
+	switch(action)
+		if("moveShuttle")
+			move_shuttle(user)
+		if("order")
+			if(world.time < reqtime)
+				visible_message("<b>[src]</b>'s monitor flashes, \"[world.time - reqtime] seconds remaining until another requisition form may be printed.\"")
+				return
+			var/amount = 1
+			if(params["multiple"] == "1") // 1 is a string here. DO NOT MAKE THIS A BOOLEAN YOU DORK
+				var/num_input = input(user, "Amount", "How many crates? (20 Max)") as null|num
+				if(!num_input || (!is_public && !is_authorized(user)) || ..()) // Make sure they dont walk away
+					return
+				amount = clamp(round(num_input), 1, 10)
+			var/datum/supply_packs/P = locateUID(params["crate"])
+			if(!istype(P))
+				return
+			var/timeout = world.time + (60 SECONDS) // If you dont type the reason within a minute, theres bigger problems here
+			var/reason = input(user, "Reason", "Why do you require this item?","") as null|text
+			if(world.time > timeout || !reason || (!is_public && !is_authorized(user)) || ..())
+				// Cancel if they take too long, they dont give a reason, they aint authed, or if they walked away
+				return
+			reason = sanitize(copytext(reason, 1, MAX_MESSAGE_LEN))
+
+			//===orderee identification information===
+			var/idname = "*None Provided*"
+			var/idrank = "*None Provided*"
+			if(ishuman(user))
+				var/mob/living/carbon/human/H = user
+				idname = H.get_authentification_name()
+				idrank = H.get_assignment()
+			else if(issilicon(user))
+				idname = user.real_name
+
+			//===orderee account information===
+			var/datum/money_account/selected_account = locateUID(params["account"])
+			for(var/i in 1 to amount)
+				var/datum/supply_order/order = SSeconomy.generate_supply_order(params["crate"], idname, idrank, amount, reason)
```

Comment:
Is this correct? You order N crates N times. Doesn't really make sense to me

Reply:
I'll look into this more, this is the consequence of me fixing the multiple order button

Reply:
fixed in next commit